src/parser/WDL/parse.js

```diff
@@ -57,7 +59,375 @@ function logicParsingStage(ast, data) {
+    return wf;
+  });
+}
+
+function getNounCalls(calls, imports) {
+  const nounCalls = {};
+  const importNames = imports.map(imp => imp.name);
+
+  calls.filter(call => importNames.includes(call.split(Constants.NS_SPLITTER)[0]))
+    .forEach((call) => {
+      const [nsName, ...callNameRest] = call.split(Constants.NS_SPLITTER);
+      const chosenImport = imports.filter(imp => imp.name === nsName).shift();
+
+      if (chosenImport) {
+        const callName = callNameRest.join(Constants.NS_CONNECTOR);
+        if (!nounCalls[nsName]) {
+          nounCalls[nsName] = {};
+        }
+        nounCalls[nsName].importObject = chosenImport;
+
+        if (nounCalls[nsName].callNames && _.isArray(nounCalls[nsName].callNames)) {
+          nounCalls[nsName].callNames.push(callName);
+        } else {
+          nounCalls[nsName].callNames = [callName];
+        }
+      }
+    });
+
+  return nounCalls;
+}
+
+/** Return array of ast tasks/workflows */
+function resolveCalls(calls, imports, preparedSubWdl) {
+  const nounCalls = getNounCalls(calls, imports);
+
+  const foundTasks = [];
+  Object.entries(nounCalls).forEach((entry) => {
+    const [nsName, v] = entry;
+    const hermesRes = hermesStage(preparedSubWdl[v.importObject.uri]);
+    const importAst = hermesRes.status ? hermesRes.ast : null;
+
+    if (!importAst) { return; }
+    // find tasks
+    importAst.attributes.body.list
+      .filter(item => item.name.toLowerCase() === Constants.TASK
+        && v.callNames.includes(item.attributes.name.source_string))
+      .forEach((task) => {
+        task.attributes.name.source_string = `${nsName}_${task.attributes.name.source_string}`;
+        foundTasks.push(task);
+      });
+
+    // and find workFlows
+    getWorkflows(importAst)
+      .filter(item => v.callNames.includes(item.attributes.name.source_string))
+      // convert workflow to task to enable it's presentation
+      .forEach((wf) => {
+        wf.attributes.name.source_string = `${nsName}_${wf.attributes.name.source_string}`;
+        // clear sub wf calls and outputs expressions
+        wf = clearWorkflowCallsAndOutputs([wf])[0];
+        foundTasks.push(wf);
+      });
+  });
+
+  return foundTasks;
+}
+
+function addSubWorkflows(ast, importedAst) {
+  const resAst = ast;
+  importedAst.forEach(item => resAst.attributes.body.list.push(item));
+
+  return resAst;
+}
+
+/** Parse function with WDL imports support */
+async function importParsingStage(firstAst, opts) {
+  const result = {
+    status: true,
+    message: '',
+    ast: firstAst,
+  };
+
+  // list parsed import statements in ast
+  const imports = getImports(firstAst);
+
+  if (!imports.length) {
+    return result;
+  }
+
+  // find calls in firstAst
+  let calls = getCallsNames(getWorkflows(firstAst));
+
+  const tasks = getTaskNames(firstAst);
+
+  // check if calls're already in existing tasks
+  calls = calls.filter(call => !tasks.includes(call));
+  if (!calls.length) return result;
```

Comment:
Let's stick to using one statement per line.

Reply:
Fixed